sentence_transformers/sampler.py

```diff
@@ -0,0 +1,127 @@
+from itertools import accumulate, cycle
+from typing import List, TYPE_CHECKING
+import logging
+
+from torch.utils.data import BatchSampler, SubsetRandomSampler
+import torch
+
+if TYPE_CHECKING:
+    from sentence_transformers.trainer import SentenceTransformerTrainer
+
+logger = logging.getLogger(__name__)
+
+
+class RoundRobinBatchSampler:
```

Comment:
for the sampler classes, shall we inherit from the torch data to make more torch native? most likely you only need a constructor and `__iter__` method

Reply:
I addressed this in d1fa120b78cd993b07aa15a3232b17a30a5c42da. I checked, and I do have to override the get_epoch, __len__ methods also, so this class ends up overriding all methods of a BatchSampler.